modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -67,21 +67,23 @@ def validate_form_526_change_of_address_required_fields!
       def validate_form_526_change_of_address_beginning_date!
         change_of_address = form_attributes['changeOfAddress']
         date = change_of_address.dig('dates', 'beginDate')
-        return unless 'TEMPORARY'.casecmp?(change_of_address['typeOfAddressChange'])
 
         # If the date parse fails, then fall back to the InvalidFieldValue
         begin
-          return if Date.strptime(date, '%m-%d-%Y') < Time.zone.now
+          Date.strptime(date, '%m-%d-%Y')
         rescue
           raise ::Common::Exceptions::InvalidFieldValue.new('changeOfAddress.dates.beginDate', date)
         end
-
-        raise ::Common::Exceptions::InvalidFieldValue.new('changeOfAddress.dates.beginDate', date)
       end
 
       def validate_form_526_change_of_address_ending_date!
         change_of_address = form_attributes['changeOfAddress']
         date = change_of_address.dig('dates', 'endDate')
+        if 'PERMANENT'.casecmp?(change_of_address['typeOfAddressChange']) && date.present?
+          raise ::Common::Exceptions::UnprocessableEntity.new(
+            detail: "\"#{date}\" is not a valid value for \"changeOfAddress.dates.endDate\""
+          )
+        end
         return unless 'TEMPORARY'.casecmp?(change_of_address['typeOfAddressChange'])
```

Comment:
can we be more specific? If the change of address is permanent, and an end date is provided, I feel like the current message is confusing, and would lead me to believe my date is incorrect rather than not needed.